### DISCUSSION TITLE
OrchardCore Architecture and Design

### DISCUSSION THREAD

**Discussion by bryp** [OrchardCMS/OrchardCore#8531] [General]
Hi folks,

I'm writing today to ask if there is any documentation available that describes the architecture and design of Orchard Core.  The Orchard Core Documentation (docs.orchardcore.net) is an excellent site that describes how to get started, how to guides, reference materials, resources, and tons of content describing how to use Orchard Core to build module applications, etc.  I get it.

But, I'm not looking for docs about how to use Orchard Core.  I'm looking for docs that describe the architecture and design of Orchard Core; that is, how Orchard Core **works** - not how to **use** Orchard Core.  Thank in advance.

**Comment by dlidstrom** [comment]
There is some documentation for Orchard Cms that seems to still be applicative for Orchard Core. For example, placement is explained very well for Orchard Cms, while being quite technically explained for Orchard Core. To understand the **why** of placement, [this page](https://docs.orchardcore.net/projects/O1/en/latest/Documentation/Understanding-placement-info/) tells it quite nicely. I suspect there is a lot more hidden among those docs.
Also, [here](https://docs.microsoft.com/en-us/archive/msdn-magazine/2012/january/orchard-cms-orchard-extensibility) is an old MSDN article from one of the founders of Orchard. It gives the background on its design and I think much of the code samples are still (mostly?) valid. That kind of background information is great IMO, and it would have been nice to have it as part of an OC curriculum.
